[torch.Package/Torchscript] rewrite ScriptModuleSerializer exporter

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54894 [torch.Package/TorchScript] TS into torch.Package test cases
* #54893 [torch.Package/TorchScript] torch.Package python logic to save TorchScript
* **#55765 [torch.Package/Torchscript] rewrite ScriptModuleSerializer exporter**
* #54891 [torch.Package/TorchScript] TS serialization importer to handle unified format

Rewrote to handle both original and unified format (I need to separate the refactoring from addition of new logic, but basically the new logic is contained to the `ScriptModuleSerializer::serialize_unified_format` method)

Commit:
[torch.Package/Torchscript] rewrite ScriptModuleSerializer exporter to
handle unified format

[ghstack-poisoned]

Commit:
Update on "[torch.Package/Torchscript] rewrite ScriptModuleSerializer exporter to"


handle unified format

[ghstack-poisoned]

Lilyjjo commented on torch/csrc/jit/serialization/export_module.cpp:
```diff
@@ -372,232 +372,270 @@ void SetExportModuleMobileInfoConverter(
-      debug_info_elements->emplace_back(
-          static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
-    }
+std::tuple<uint64_t, uint64_t> ScriptModuleSerializer::serialize_unified_format(
```
This method is the only read 'new' logic in this PR, the rest of it is refactoring of the already existing logic

Commit:
Update on "[torch.Package/Torchscript] rewrite ScriptModuleSerializer exporter"


Rewrote to handle both original and unified format (I need to separate the refactoring from addition of new logic, but basically the new logic is contained to the `ScriptModuleSerializer::serialize_unified_format` method)

[ghstack-poisoned]
